src/Integration/MefServices/ActiveSolutionTracker.cs

```diff
@@ -170,7 +174,16 @@ private void RaiseSolutionChangedEvent(bool isSolutionOpen)
         {
             // Note: if lightweight solution load is enabled then the solution might not
             // be fully opened at this point
-            ActiveSolutionChanged?.Invoke(this, new ActiveSolutionChangedEventArgs(isSolutionOpen));
+            if (isSolutionOpen)
```

Comment:
The `if` is not necessary, the solutionInfoProvider is taking care to return null in case there isn't an open solution.

Reply:
Optimization. We don't need to call SolutionInfoProvider. It will save us 1 thread check and vs hierarchy lookup